great_tables/data/__init__.py

```diff
@@ -1,1355 +1,1488 @@
-    "value": "float64",
-    "uncert": "float64",
-    "sf_value": "Int64",
-    "sf_uncert": "Int64",
-    "units": "object",
-}
-
-_illness_fname = DATA_MOD / "13-illness.csv"
-_illness_dtype = {
-    "test": "object",
-    "units": "object",
-    "day_3": "float64",
-    "day_4": "float64",
-    "day_5": "float64",
-    "day_6": "float64",
-    "day_7": "float64",
-    "day_8": "float64",
-    "day_9": "float64",
-    "norm_l": "float64",
-    "norm_u": "float64",
-}
-
-_reactions_fname = DATA_MOD / "14-reactions.csv"
-_reactions_dtype = {
-    "cmpd_name": "object",
-    "cmpd_mwt": "float64",
-    "cmpd_formula": "object",
-    "cmpd_type": "object",
-    "cmpd_smiles": "object",
-    "cmpd_inchi": "object",
-    "cmpd_inchikey": "object",
-    "OH_k298": "float64",
-    "OH_uncert": "float64",
-    "OH_u_fac": "float64",
-    "OH_A": "float64",
-    "OH_B": "float64",
-    "OH_n": "float64",
-    "OH_t_low": "float64",
-    "OH_t_high": "float64",
-    "O3_k298": "float64",
-    "O3_uncert": "float64",
-    "O3_u_fac": "float64",
-    "O3_A": "float64",
-    "O3_B": "float64",
-    "O3_n": "float64",
-    "O3_t_low": "float64",
-    "O3_t_high": "float64",
-    "NO3_k298": "float64",
-    "NO3_uncert": "float64",
-    "NO3_u_fac": "float64",
-    "NO3_A": "float64",
-    "NO3_B": "float64",
-    "NO3_n": "float64",
-    "NO3_t_low": "float64",
-    "NO3_t_high": "float64",
-    "Cl_k298": "float64",
-    "Cl_uncert": "float64",
-    "Cl_u_fac": "float64",
-    "Cl_A": "float64",
-    "Cl_B": "float64",
-    "Cl_n": "float64",
-    "Cl_t_low": "float64",
-    "Cl_t_high": "float64",
-}
-
-_photolysis_fname = DATA_MOD / "15-photolysis.csv"
-_photolysis_dtype = {
-    "cmpd_name": "object",
-    "cmpd_formula": "object",
-    "products": "object",
-    "type": "object",
-    "l": "float64",
-    "m": "float64",
-    "n": "float64",
-    "quantum_yield": "float64",
-    "wavelength_nm": "object",
-    "sigma_298_cm2": "object",
-}
-
-_nuclides_fname = DATA_MOD / "16-nuclides.csv"
-_nuclides_dtype = {
-    "nuclide": "object",
-    "z": "Int64",
-    "n": "Int64",
-    "element": "object",
-    "radius": "float64",
-    "radius_uncert": "float64",
-    "abundance": "float64",
-    "abundance_uncert": "float64",
-    "is_stable": "object",
-    "half_life": "float64",
-    "half_life_uncert": "float64",
-    "isospin": "object",
-    "decay_1": "object",
-    "decay_1_pct": "float64",
-    "decay_1_pct_uncert": "float64",
-    "decay_2": "object",
-    "decay_2_pct": "float64",
-    "decay_2_pct_uncert": "float64",
-    "decay_3": "object",
-    "decay_3_pct": "float64",
-    "decay_3_pct_uncert": "float64",
-    "magnetic_dipole": "float64",
-    "magnetic_dipole_uncert": "float64",
-    "electric_quadrupole": "float64",
-    "electric_quadrupole_uncert": "float64",
-    "atomic_mass": "float64",
-    "atomic_mass_uncert": "float64",
-    "mass_excess": "float64",
-    "mass_excess_uncert": "float64",
-}
+
+class _DataFrameConverter:
```

Comment:
In case it's useful, I ended up creating a tiny DataFrame like implementation in reactable-py, just so I could feed example data for demos

https://github.com/machow/reactable-py/blob/main/reactable/simpleframe.py

I think a nice advantage of your approach here though is that it doesn't read the csv until you use one of the `to_*()` methods!

Reply:
The code looks great! It seems you'll need to keep the data as an internal variable within `simpleframe` to enable `get` and `set` actions. However, for our dataset use case, we should be able to store just the `filepath` and `dtype` rather than holding the entire dataset in memory.